testsuite/MDAnalysisTests/transformations/test_nojump.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+import MDAnalysis as mda
+from MDAnalysis.transformations import NoJump
+from MDAnalysisTests import datafiles as data
+
+
+@pytest.fixture()
+def nojump_universe():
+    """
+    Create the universe objects for the tests.
+    """
+    u = mda.Universe.empty(N, trajectory=True)
+    coordinates = np.empty((100,  # number of frames
+                            u.atoms.n_atoms,
+                            3))
+    coordinates[::3,0] = 0 * np.ones(3)
+    coordinates[1::3,0] = 0.3 * np.ones(3)
+    coordinates[2::3,0] = 0.6 * np.ones(3)
+    u.load_new(coordinates, order='fac')
+    return u
+
+
+def test_nojump_orthogonal_fwd(nojump_universe):
+    """
+    Test if the nojump transform is returning the correct
+    values when iterating forwards over the sample trajectory.
+    """
+    u = nojump_universe
+    dim = np.asarray([1, 1, 1, 90, 90, 90], np.float32)
+    workflow = [mda.transformations.boxdimensions.set_dimensions(dim), NoJump()]
+    u.add_transformations(*workflow)
+    transformed_coordinates = u.trajectory.timeseries()[0]
+    # Step is 1 unit every 3 steps. After 99 steps from the origin,
+    # we'll end up at 33.
+    assert_allclose(transformed_coordinates[99], 33 * np.ones(3))
+
+
+def test_nojump_nonorthogonal_fwd(nojump_universes_nonorthogonal):
+    """
+    Test if the nojump transform is returning the correct
+    values when iterating forwards over the sample trajectory.
+    """
+    u = nojump_universe
+    dim = np.asarray([1, 1, 1, 90, 60, 90], np.float32)
+    workflow = [mda.transformations.boxdimensions.set_dimensions(dim), NoJump()]
+    u.add_transformations(*workflow)
+    transformed_coordinates = u.trajectory.timeseries()[0]
+    # Step is 1 unit every 3 steps. After 99 steps from the origin,
+    # we'll end up at 33. However, since the unit cell is non-orthogonal,
+    # we'll end up at a distorted place.
+    assert_allclose(transformed_coordinates[99], 33 * np.array([0.5, 1, np.sqrt(3)/2]))
```

Comment:
Same here, match the whole array.
- Do we know the exact for of the skew factor `33 * np.array([0.5, 1, np.sqrt(3)/2]` is correct? If so I would love to see the working in the code.

Reply:
Oof. This is a pain to do for how I constructed the non-orthogonal case. The z position shows this the most clearly.

```
[[ 0.          0.          0.        ]
 [ 0.33333334  0.33333334  0.33333334]
 [ 0.6666667   0.6666667   0.6666667 ]
 [ 0.5         1.          0.8660254 ]
 [ 0.8333334   1.3333334   1.1993587 ]
 [ 1.1666667   1.6666667   1.5326921 ]
 [ 1.          2.          1.7320508 ]
...snip...
```

The jumps aren't all the same size, since the skewed box means that a "constant" displacement in wrapped space meets up with the next box over in a weird place. Its too late at night to think about how this works.

Reply:
Feel free to change your box vectors freely to make the working easier if it helps. 😀

Reply:
Yeah, except that you REALLY want to test on non-orthogonal boxes, since its harder. The path I chose is just inherently jagged. I've made another test based on a straight line.